skfp/descriptors/constitutional.py

```diff
@@ -0,0 +1,252 @@
+from typing import Optional, Union
+
+from rdkit.Chem import AddHs, Mol
+from rdkit.Chem.Descriptors import MolWt
+from rdkit.Chem.rdMolDescriptors import CalcNumRings, CalcNumRotatableBonds
+
+
+def validate_molecule(func):
+    """
+    Decorator to validate that the molecule has at least one atom.
+
+    Raises
+    ------
+    ValueError
+        If the molecule has no atoms.
+    """
+
+    def wrapper(mol: Mol, *args, **kwargs):
```

Comment:
You should use `functools.wraps()` also, see https://realpython.com/primer-on-python-decorators/ and https://docs.python.org/3/library/functools.html#functools.wraps